File: src/MyServiceBus/NamingConventions.cs
using System;
using System.Reflection;

namespace MyServiceBus;

public static class NamingConventions
{
    static IMessageEntityNameFormatter _entityNameFormatter = new DefaultMessageEntityNameFormatter();

    public static IMessageEntityNameFormatter EntityNameFormatter => _entityNameFormatter;

    public static void SetEntityNameFormatter(IMessageEntityNameFormatter formatter)
    {
        _entityNameFormatter = formatter;
    }

    public static string GetMessageUrn(Type messageType)
    {
        return $"urn:message:{GetMessageName(messageType)}";
    }

    [Throws(typeof(AmbiguousMatchException))]
    public static string GetExchangeName(Type messageType)
    {
        var attr = messageType.GetCustomAttribute<EntityNameAttribute>();
        if (attr != null)
            return attr.EntityName;

        return _entityNameFormatter.FormatEntityName(messageType);
    }

    public static string GetMessageName(Type messageType)
    {
        return $"{messageType.Namespace}:{messageType.Name}";
    }

    public static string GetQueueName(Type messageType)
    {
        return $"{messageType.Name.ToLower().Replace('.', '-')}2-consumer";
    }

    class DefaultMessageEntityNameFormatter : IMessageEntityNameFormatter
    {
        public string FormatEntityName(Type messageType)
        {
            return GetMessageName(messageType);
        }
    }
}

File: src/Java/servicebus/src/main/java/com/myservicebus/NamingConventions.java
package com.myservicebus;

public class NamingConventions {

    public static String getMessageUrn(Class<?> messageType) {
        return String.format("urn:message:%s", getMessageName(messageType));
    }

    public static String getExchangeName(Class<?> messageType) {
        return getMessageName(messageType);
    }

    public static String getMessageName(Class<?> messageType) {
        return String.format("TestApp:%s", messageType.getSimpleName());
    }

    public static String getQueueName(Class<?> consumerType) {
        return consumerType.getName().toLowerCase().replace(".", "-") + "-consumer";
    }
}
